src/enclave/Enclave/Join.cpp

```diff
@@ -92,11 +96,13 @@ void non_oblivious_sort_merge_join(
               + to_string(current));
           }
 
+          EnclaveContext::getInstance().set_append_mac(true);
           w.append(primary, current);
         }
       }
     }
   }
 
-  w.output_buffer(output_rows, output_rows_length);
+  EnclaveContext::getInstance().set_append_mac(true);
```

Comment:
`append_mac` is default true, so unless we at some point in aggregation set it to be false, we won't have to call this.

We do set `append_mac` to false at times during because we may call `output_buffer()` on blocks that aren't the result of the ecall, i.e. they're intermediate results of the operator and will not be received as input by a subsequent operator. We don't do integrity checks on these for now, but we should likely add this in later. (I made a note about this in the multiparty timeline doc)